jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/event/CommentNotificationEventListenerTest.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.casemgmt.impl.event;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.jbpm.casemgmt.impl.event.CommentNotificationEventListener.AUTHOR_PARAM;
+import static org.jbpm.casemgmt.impl.event.CommentNotificationEventListener.CASE_ID_PARAM;
+import static org.jbpm.casemgmt.impl.event.CommentNotificationEventListener.COMMENT_ID_PARAM;
+import static org.jbpm.casemgmt.impl.event.CommentNotificationEventListener.COMMENT_PARAM;
+import static org.jbpm.casemgmt.impl.event.CommentNotificationEventListener.CREATED_AT_PARAM;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.jbpm.casemgmt.api.event.CaseCommentEvent;
+import org.jbpm.casemgmt.api.model.CaseRole;
+import org.jbpm.casemgmt.api.model.instance.CaseFileInstance;
+import org.jbpm.casemgmt.impl.model.CaseRoleImpl;
+import org.jbpm.casemgmt.impl.model.instance.CaseFileInstanceImpl;
+import org.jbpm.casemgmt.impl.model.instance.CommentInstanceImpl;
+import org.jbpm.casemgmt.impl.util.TestNotificationPublisher;
+import org.junit.Test;
+import org.kie.api.task.model.OrganizationalEntity;
+import org.kie.api.task.model.User;
+
+public class CommentNotificationEventListenerTest {
+
+    @Test
+    public void testFindMentionsNoMentionsInComment() {
```

Comment:
Very nice example of unit test - this is exactly where it makes complete sense. One suggestion - rather a matter of preference - extracting the pattern into a standalone (maybe static utility) class e.g. `UserMentionMatcher` could make unit testing even 1 step simpler - the class would provide a public method. `CommentNotificationEventListener` could just use the class and hide the `extractMentionedRoles` method as private. In theory, `UserMentionMatcher` could be reused somewhere else later. But this suggestion is rather nitpicking. Thanks for the unit coverage!